control-plane/agents/src/bin/core/controller/reconciler/nexus/mod.rs

```diff
@@ -164,18 +164,19 @@ async fn handle_child_rebuild(
 ) -> Result<(), SvcError> {
     let wait_duration = RuleSet::faulted_child_wait(nexus, context.registry());
     let is_elapsed = wait_duration_elapsed(child.faulted_at, wait_duration, &child.uri);
-    if child.state_reason == ChildStateReason::RebuildFailed || is_elapsed {
-        info!(%child.uri, "Start full rebuild for child, elapsed: {}, child state reason: {:?}", is_elapsed.to_string(), child.state_reason);
-        faulted_children_remover(nexus_spec, child, context).await?
-    } else if get_child_replica(nexus_spec.as_ref(), child, context)
-        .await
-        .is_ok()
+    if (child.state_reason.clone() != ChildStateReason::RebuildFailed)
```

Comment:
Can we move RebuildFailed to a separate condition so we don't have to keep checking?
```suggestion
    if (child.state_reason == ChildStateReason::RebuildFailed) { return fault... }
```

Reply:
or take it in a variable in the beginning like is_elapsed:
`let is_rebuild_failed = (child.state_reason == ChildStateReason::RebuildFailed);`